addons/velero/1.6.2/install.sh

```diff
@@ -162,13 +227,142 @@ function velero_patch_http_proxy() {
+        VELERO_S3_ACCESS_KEY_SECRET=$(kubectl -n minio get secret minio-credentials -ojsonpath='{ .data.MINIO_SECRET_KEY }' | base64 --decode)
+    fi
+
+    # If this is run through `velero_already_applied`, we need to create base kustomization file
+    if [ ! -f "$dst/kustomization.yaml" ];then
+        render_yaml_file "$src/tmpl-kustomization.yaml" > "$dst/kustomization.yaml"
+    fi
+
+    # TODO (dans): figure out if there is enough space create a new volume with all the snapshot data
+
+    # create secret for migration init container to pull from object store
+    render_yaml_file "$src/tmpl-s3-migration-secret.yaml" > "$dst/s3-migration-secret.yaml"
+    insert_resources "$dst/kustomization.yaml" s3-migration-secret.yaml
+
+    # create configmap that holds the migration script
+    render_yaml_file "$src/tmpl-s3-migration-configmap.yaml" > "$dst/s3-migration-configmap.yaml"
+    insert_resources "$dst/kustomization.yaml" s3-migration-configmap.yaml
+
+    # add patch to add init container for migration
+    render_yaml_file "$src/tmpl-s3-migration-deployment-patch.yaml" > "$dst/s3-migration-deployment-patch.yaml"
+    insert_patches_strategic_merge "$dst/kustomization.yaml" s3-migration-deployment-patch.yaml
+
+    # update the BackupstorageLocation
+    render_yaml_file "$src/tmpl-s3-migration-bsl.yaml" > "$dst/s3-migration-bsl.yaml"
+    insert_resources "$dst/kustomization.yaml" s3-migration-bsl.yaml
+
+    export WILL_MIGRATE_VELERO_OBJECT_STORE="1"
+}
+
+# add patches for the velero and restic to the current kustomization file that setup the PVC setup like the 
+# velero LVP plugin requires 
+function velero_patch_internal_pvc_snapshots() {
+    local src="$1"
+    local dst="$2"
+
+    # If we are migrating from Rook to Longhorn, longhorn is not yes the default storage class.
+    export VELERO_PVC_STORAGE_CLASS="default" # this is the rook-ceph default storage class
+    if [ -n "$LONGHORN_VERSION" ]; then
+        export VELERO_PVC_STORAGE_CLASS="longhorn"
+    fi
+
+    # create the PVC
+    render_yaml_file "$src/tmpl-internal-snaps-pvc.yaml" > "$dst/internal-snaps-pvc.yaml"
+    insert_resources "$dst/kustomization.yaml" internal-snaps-pvc.yaml
+
+    # add patch to add the pvc in the correct location for the velero deployment
+    render_yaml_file "$src/tmpl-internal-snaps-deployment-patch.yaml" > "$dst/internal-snaps-deployment-patch.yaml"
+    insert_patches_strategic_merge "$dst/kustomization.yaml" internal-snaps-deployment-patch.yaml
+
+    # add patch to add the pvc in the correct location for the restic daemonset
+    render_yaml_file "$src/tmpl-internal-snaps-ds-patch.yaml" > "$dst/internal-snaps-ds-patch.yaml"
+    insert_patches_strategic_merge "$dst/kustomization.yaml" internal-snaps-ds-patch.yaml
+
+}
+
+function velero_pvc_bound() {
+    kubectl get pvc velero-internal-snapshots -n ${VELERO_NAMESPACE} -oyaml | grep -q "phase: Bound"
```

Comment:
better to use templates here?

Reply:
How are you proposing to use templates in this case? The problem in my mind was that I can't patch the PV until it has been created for the PVC, so this is a hack to know whether there is a corresponding PVC/PV before I try to grab the PV name.

Reply:
I meant the flag to the kubectl comment to extract the phase.

Reply:
ah, ok. I can do that.